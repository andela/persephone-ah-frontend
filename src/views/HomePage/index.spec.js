import React from 'react';
import { mount } from 'enzyme';
import { BrowserRouter } from 'react-router-dom';
import { HomePage } from './index.jsx';

const defaultProps = {
  themeToggler: jest.fn(),
  theme: {}
};

describe('Render component', () => {
  it('should render component successfully', () => {
    const component = mount(
      <BrowserRouter>
        <HomePage {...defaultProps} />{' '}
      </BrowserRouter>
    );
    expect(
      component.contains(
        <h2 className="font-weight-light">
          Find the best resources on tech related articles
        </h2>
      )
    ).toBeTruthy();
  });

  it('should toggle to light theme', () => {
    const component = mount(
      <BrowserRouter>
        <HomePage {...defaultProps} />
      </BrowserRouter>
    );
    expect(component.find('button')).toBeTruthy();
    expect(component).toMatchSnapshot();
  });
});
