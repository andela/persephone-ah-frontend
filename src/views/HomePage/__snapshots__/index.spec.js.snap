// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render component should toggle to light theme 1`] = `ReactWrapper {}`;
